htdocs/core/class/commonobject.class.php

```diff
@@ -4635,7 +4635,7 @@ function showOptionals($extrafields, $mode='view', $params=null, $keyprefix='')
 					case "edit":
 						// GETPOST("options_" . $key) can be 'abc' or array(0=>'abc')
 						$getposttemp = GETPOST('options_'.$key, 'none');				// GETPOST can get value from GET, POST or setup of default values.
-						if (isset($getposttemp)) {
+						if (isset($getposttemp) && $getposttemp != '') {
```

Comment:
What if getposttemp is an array ?
Also which kind of bug does this fix ?

Reply:
@eldy 
add an extrafields in thirdparty by example, edit a thirdparty and add value in field, valid... the value is visible in view mode,  but edit again this thirparty, the value is hidden in edit mode